components/CheckDocumentation/vacuum/XminHorizon.tsx

```diff
@@ -0,0 +1,205 @@
+import SQL from "../../SQL";
+import React from "react";
+
+import {
+  CheckDocs,
+  CheckGuidanceProps,
+  CheckTriggerProps,
+} from "../../../util/checks";
+import { useCodeBlock } from "../../CodeBlock";
+
+const XminHorizonTrigger: React.FunctionComponent<CheckTriggerProps> = ({
+  config,
+}) => {
+  const threshold = config.settings["behind_days"] as number;
+  const dayPluralized = threshold === 1 ? "day" : "days";
+  return (
+    <p>
+      Detects when the xmin horizon on the server has not progressed for the
+      last <code>{threshold}</code> {dayPluralized} and creates an issue with
+      severity "info". The issue will be created even if no VACUUM is currently
+      blocked by this, as this will potentially block any future VACUUMs.
+      Resolves once the xmin horizon makes some progress.
+    </p>
+  );
+};
+
+const XminHorizonGuidance: React.FunctionComponent<CheckGuidanceProps> = ({
+  issue,
+}) => {
+  const CodeBlock = useCodeBlock();
+  const issueDetails = issue && JSON.parse(issue.detailsJson);
+  const heldBackBy = issueDetails && issueDetails["held_back_by"];
+  let byBackend = null;
+  let byReplicationSlot = null;
+  let byReplicationSlotCatalog = null;
+  let byStandby = null;
+  let byPreparedXact = null;
+  if (heldBackBy) {
+    byBackend = heldBackBy.find((v) => v["type"] === "backend");
+    byReplicationSlot = heldBackBy.find(
+      (v) => v["type"] === "replication_slot"
+    );
+    byReplicationSlotCatalog = heldBackBy.find(
+      (v) => v["type"] === "replication_slot_catalog"
+    );
+    byStandby = heldBackBy.find((v) => v["type"] === "standby");
+    byPreparedXact = heldBackBy.find((v) => v["type"] === "prepared_xact");
+  }
+  return (
+    <div>
+      <h4>Impact</h4>
+      <p>VACUUM is potentially blocked by the xmin horizon.</p>
+      <p>
+        The xmin horizon tells you until which point the vacuum process can
+        clean up dead rows. When this value is behind and not making progress,
+        it's very likely that VACUUM can't clean up dead rows, hence blocking
+        VACUUM.
+      </p>
+      <p>
+        When VACUUM is blocked and dead rows can't be cleaned, it can result to
+        the unnecessary table bloat or slow queries.
+      </p>
+      <h4>Common Causes</h4>
+      <ul>
+        <li>
+          <h5>Long-running transactions</h5>
+          <p>
+            When there is a long running transaction, you can't clean up rows
+            that the transaction can see, even if it already became a dead row.
+          </p>
+        </li>
+        <li>
+          <h5>Delayed/bad replication slots</h5>
```

Comment:
```suggestion
          <h5>Lagging or abandoned replication slots</h5>
```
Mostly a personal preference thing, but I don't like slashes in headings like this. And I think "lagging" is slightly more accurate than "delayed"  here, and "bad" is pretty vague. "Stale" could be another alternative to "abandoned".

Reply:
+1. I like stale more so I'll go with stale 👍